climada/test/test_cropyield_integr.py

```diff
@@ -0,0 +1,165 @@
+        if_cp.check()
+
+        impact = Impact()
+        impact.calc(exp, if_cp, haz, save_mat=True)
+
+        exp_nan = CropProduction()
+        exp_nan.set_from_single_run(input_dir=INPUT_DIR, filename=FILENAME_LU, hist_mean=FILENAME_MEAN,
+                                              bbox=[0, 42, 10, 52], yearrange=(2001, 2005),
+                                              scenario='flexible', unit='t', crop='whe', irr='firr')
+        exp_nan.value[exp_nan.value==0] = np.nan
+        exp_nan.assign_centroids(haz, threshold=20)
+
+        impact_nan = Impact()
+        impact_nan.calc(exp_nan, if_cp, haz, save_mat=True)
+        self.assertListEqual(list(impact.at_event), list(impact_nan.at_event))
+        self.assertAlmostEqual(12.056545220060798, impact_nan.aai_agg)
+        self.assertAlmostEqual(12.056545220060798 , impact.aai_agg)
+
+    def test_hist_mean_of_full_haz_set(self):
+        """Test creation of full hazard set"""
+
+        output_list = list()
+        bbox = [116.25, 38.75,  117.75, 39.75]
+        files_his = ['gepic_gfdl-esm2m_ewembi_historical_2005soc_co2_yield-whe-noirr_global_DEMO_TJANJIN_annual_1861_2005.nc',
+                     'pepic_miroc5_ewembi_historical_2005soc_co2_yield-whe-firr_global_annual_DEMO_TJANJIN_1861_2005.nc',
+                     'pepic_miroc5_ewembi_historical_2005soc_co2_yield-whe-noirr_global_annual_DEMO_TJANJIN_1861_2005.nc']
+        
+        (his_file_list, file_props, hist_mean_per_crop, 
+          scenario_list, crop_list) = init_hazard_set(files_his, input_dir=INPUT_DIR,
+                                                      bbox=bbox, isimip_run = 'test_file', 
+                                                      yearrange_his = np.array([1980,2005]))
+        yearrange_mean = np.array([1980,2005])
+        for his_file in his_file_list:
+            haz_his, filename, hist_mean = calc_his_haz(his_file, file_props, input_dir=INPUT_DIR, 
+                                                        bbox=bbox, yearrange_mean=yearrange_mean)
+
+            hist_mean_per_crop[(file_props[his_file])['crop_irr']]['value'][ 
+                hist_mean_per_crop[(file_props[his_file])['crop_irr']]['idx'], :] = hist_mean
+            hist_mean_per_crop[file_props[his_file]['crop_irr']]['idx'] += 1
+        
+        
+        self.assertEqual(np.shape(hist_mean_per_crop['whe-firr']['value'])[0], 1)
+        self.assertEqual(np.shape(hist_mean_per_crop['whe-noirr']['value'])[0], 2) 
+        
+        # calculate mean hist_mean for each crop-irrigation
+        for crop_irr in crop_list:
+            mean = np.mean((hist_mean_per_crop[crop_irr])['value'], 0)
+            output_list.append(mean)
+        
+        self.assertEqual('whe-noirr', crop_list[0])
+        self.assertEqual(np.mean(hist_mean_per_crop['whe-noirr']['value']), np.mean(output_list[0]))
+        self.assertEqual(np.mean(hist_mean_per_crop['whe-noirr']['value'][:,1]), output_list[0][1])
+        
```

Comment:
Some methods seem to be missing a unit test such as `calc_fut_haz`

Reply:
yes we excluded it because all parts of it are tested in the unit tests and we did not want to add more test files to the repositories

Reply:
Fair enough. It is however a quite dangerous route to take as your test is now an interdependent function test.